### PR TITLE
Fix clippy warnings

### LINKED ISSUE #59: rust toolchain Improvements and fix clippy
- [x] add rustfmt.toml: https://github.com/encointer/encointer-node/blob/master/rustfmt.toml
- [x] upgrade rust-toolchain to rust-toolchain.toml: https://github.com/integritee-network/parachain/blob/master/rust-toolchain.toml
- [ ] fix clippy warnings

### PR DESCRIPTION
**Pull request by pifragile**
* Closes #59

### CODE REVIEW

File: ceremonies/src/lib.rs [hunk 1]
```diff
@@ -1266,37 +1262,51 @@ impl<T: Config> Pallet<T> {
 
 		info!(target: LOG, "purging ceremony index {} history for {:?}", cindex, cid);
 
+		#[allow(deprecated)]
```
Comment (brenzi) [inline ceremonies/src/lib.rs:1265]: You could set this flag at the fn definition and it should be calid for the entire fn. Or do yiu want to isolate steictly?

**Review by brenzi — APPROVED** [review]
Nice cleanup

File: ceremonies/src/lib.rs (hunk 1 above)
Comment (pifragile) [reply ceremonies/src/lib.rs]: indeed this is nicer!

**Review by clangenb — COMMENTED** [review]
Some remarks, but it looks great in general!

**Review by clangenb — CHANGES_REQUESTED** [review]
Changed it to request changes so it can't be merged, after brenzi approved already. :)

File: ceremonies/src/lib.rs [hunk 2]
```diff
@@ -61,6 +58,7 @@ const LOG: &str = "encointer";
 pub use pallet::*;
 pub use weights::WeightInfo;
 
+#[allow(clippy::unused_unit)]
```
[the remark this answers is not in the document]
Comment (pifragile) [reply ceremonies/src/lib.rs:61]: yes

File: ceremonies/rpc/src/lib.rs [hunk 3]
```diff
@@ -111,11 +111,13 @@ where
...
-			None => None,
-		}
+	pub fn get_storage<V: Decode>(&self, key: &[u8]) -> RpcResult<Option<V>> {
+		self.storage
+			.read()
+			.get(STORAGE_PREFIX, key)
+			.map(|v| Decode::decode(&mut v.as_slice()))
+			.transpose()
+			.map_err(|e| Error::OffchainStorageDecodeError(e.to_string()).into())
 	}
```
Comment (clangenb) [inline ceremonies/rpc/src/lib.rs:121]: Nice!

**Review by clangenb — APPROVED** [review]
Cool, good changes here!

**Pull request merged**